catalog/app/utils/AWS/Athena.tsx

```diff
@@ -0,0 +1,31 @@
+import Athena from 'aws-sdk/clients/athena'
+import * as React from 'react'
+
+import useMemoEqLazy from 'utils/useMemoEqLazy'
+
+import * as Config from './Config'
+
+const AthenaContext = React.createContext<() => Athena | null>(() => null)
+
+type AthenaProviderProps = React.PropsWithChildren<Partial<Athena.ClientConfiguration>>
+
+export const Provider = function AthenaProvider({
+  children,
+  ...overrides
+}: AthenaProviderProps) {
+  const cfg = Config.use()
+
+  const client: () => Athena = useMemoEqLazy(
+    {
+      ...cfg,
+      ...overrides,
+    },
+    (opts: Athena.ClientConfiguration) => new Athena(opts),
+  )
+
+  return <AthenaContext.Provider value={client}>{children}</AthenaContext.Provider>
+}
+
+export const useAthena = () => React.useContext(AthenaContext)()
```

Comment:
i think we should add `Credentials.use().suspend()` here to ensure the credentials are resolved (like in S3 provider)